front: filter on exceptions and display paced train status depending on occurrences

clarani opened the pull request:
closes #12619

clarani committed:
front: display icons depending on occurrence status

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: synchronize pacedTrain status indicator

Depending on the worst status of the model train and its activated occurrence,
adapt the color of the pacedTrain's status indicator.

invalid > schedule not met > margins consumed > valid

part of #12619

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

SharglutDev commented on front/src/styles/scss/applications/operationalStudies/_pacedTrain.scss:
```diff
@@ -184,6 +179,12 @@
         flex-wrap: nowrap;
 
+        &.invalid {
+          background-image: url('data:image/svg+xml,%3Csvg version="1.1" viewBox="0 0 48 48" xmlns="http://www.w3.org/2000/svg"%3E%3Ctitle%3EAssets/TrainList/InvalidTrain%3C/title%3E%3Cg fill-rule="evenodd"%3E%3Crect width="48" height="48" fill="%23FFEEED"/%3E%3Cpath d="m48 26v22h-22l22-22zm0-26-48 48v-22l26-26h22z" fill="%23FF6868" opacity=".1"/%3E%3C/g%3E%3C/svg%3E');
```
<img width="393" height="106" alt="Capture d’écran 2025-07-31 à 08 59 41" src="https://github.com/user-attachments/assets/cc2b7316-f098-4178-b9ff-a23475ad426e" />

This is the old image, we should use the new one (you can find it l.123 in the same file) to be able to see the occurrence selection.

clarani replied:
Done

SharglutDev commented on front/src/styles/scss/applications/operationalStudies/_pacedTrain.scss:
This is case 13 on @maelysLeratRosso ppt, we should not see the clock on the paced train card (same if it was a too fast) 
<img width="402" height="208" alt="Capture d’écran 2025-07-31 à 10 08 27" src="https://github.com/user-attachments/assets/52b46b57-40aa-42bd-9e90-47b5d2934df8" />

clarani replied:
Can we challenge this with Maëlys ?

Because the AC in the main ticket is the following :) 
> Pour la carte parent de la mission :
> - le fond et les icônes correspondent au statut du train modèle
> - Le petit trait à droite (orange ou rouge) correspond au "pire" état parmi le train modèle et ses occurrences.

SharglutDev replied:
You're right, sorry for the comment !

SharglutDev commented on front/src/styles/scss/applications/operationalStudies/_pacedTrain.scss:
<img width="401" height="177" alt="Capture d’écran 2025-07-31 à 09 21 00" src="https://github.com/user-attachments/assets/d49ba57a-e2bf-407c-9b3a-e1555708e1fb" />

I think the yellow bar shoud be aligned with the occurrence main infos

clarani replied:
Done

SharglutDev commented on front/src/modules/timetableItem/components/Timetable/useFilterTimetableItems.ts:
When applying filter "partial punctuality" and "invalid items", I don't see any item in the list but I think I would expect to see paced trains which both have a not honored occurrence and invalid one ? 
Or am I wrong @maelysLeratRosso ?

maelysLeratRosso replied:
When applying filter "partial punctuality" : I expect to see all the services for which the paced train or at least one of the occurrences has partial punctuality. And for these services, I expect to see all the occurrences (even those with full punctuality)

Same for "invalid items"

SharglutDev commented on front/src/modules/timetableItem/components/Timetable/PacedTrain/OccurrenceItem.tsx:
One of the AC says `For deactivated occurrences, validity, timetable and margin status are not displayed.`, but that's not the case here :

<img width="395" height="173" alt="Capture d’écran 2025-07-31 à 10 32 18" src="https://github.com/user-attachments/assets/158951f3-8439-43ae-a7c5-8ab51b2bfb79" />

clarani replied:
Done

SharglutDev commented on front/src/modules/timetableItem/components/Timetable/useFilterTimetableItems.ts:
```diff
@@ -56,108 +58,118 @@ const useFilterTimetableItems = (
+      speedLimitTag?: string | null | undefined;
+      rollingStock?: LightRollingStock;
+      category?: TrainMainCategory | null | undefined;
+    }) => {
```
nit : maybe we could rather `Pick<TrainScheduleWithDetails, ...>` here instead one of these properties type changes in the future.

clarani replied:
Done :white_check_mark:

SharglutDev reviewed (commented):
Good job, really nice refacto for the filter ! (even though it would have been more perfect if the refacto was in a separated commit than the exception handle 🙂)

clarani committed:
front: create a new function filterTimetableItem

This function will be useful for the following commits.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: filter timetableItem by type of object outside the filterTimetableItem func

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

clarani committed:
front: apply filters on services exceptions

When filtering, the missions for which at least one occurrence matches the filter are displayed.

Signed-off-by: Clara Ni <clara.ni@outlook.fr>

SharglutDev commented on front/src/modules/timetableItem/components/Timetable/useFilterTimetableItems.ts:
```diff
@@ -171,6 +141,37 @@ const useFilterTimetableItems = (
+            : undefined;
+          return {
+            name: exception.train_name?.value ?? modelTrain.name,
+            category: exception.rolling_stock_category?.value,
```
We should fallback to the paced train category if there is no category exception

SharglutDev reviewed (approved):
Lgtm and tested, great job !

You can close the last comment when fixed

achrafmohye reviewed (approved):
LGTM thanks
